src/search/components/SearchForm.js

```diff
@@ -7,19 +7,26 @@ import 'search/css/lib/buttons.scss';
 
 export default class SearchForm extends React.Component {
   static propTypes = {
-    onSearch: PropTypes.func.isRequired,
+    pathname: PropTypes.string.isRequired,
+    query: PropTypes.string,
+  }
+  static contextTypes = {
+    router: PropTypes.object,
   }
 
   handleSubmit = (e) => {
     e.preventDefault();
-    this.props.onSearch(this.refs.query.value);
+    const { pathname } = this.props;
+    const query = this.refs.query.value;
+    this.context.router.push(`${pathname}?q=${query}`);
```

Comment:
any special reason why you aren't using [react-redux-router](https://github.com/reactjs/react-router-redux)? It will preserve all the time travel and other stateless features you get from redux.